src/saml.py

```diff
@@ -2,14 +2,22 @@
 # See LICENSE file for licensing details.
 
 """Provide the SamlApp class to encapsulate the business logic."""
+import base64
+import hashlib
 import logging
+import secrets
 import urllib.request
 from functools import cached_property
+from typing import TYPE_CHECKING, Optional
 
 from charms.saml_integrator.v0 import saml
 
 from charm_state import CharmConfigInvalidError, CharmState
 
+if TYPE_CHECKING:  # pragma: nocover
+    from lxml import etree  # nosec
```

Comment:
What is the `# nosec` here please?

Reply:
It's a blacklisted import https://bandit.readthedocs.io/en/1.6.3/blacklists/blacklist_imports.html

Reply:
[This](https://github.com/PyCQA/bandit/issues/767) can probably shed some light, along with [this lxml post](https://lxml.de/FAQ.html#how-do-i-use-lxml-safely-as-a-web-service-endpoint)

Reply:
Should we include that information in a comment? I can imagine future readers will have the same question

Reply:
Sure thing